File: python/9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        else:
            num = x
            reverse = 0
            while num != 0:
                reverse = reverse * 10 + num % 10
                num = num // 10
            if x == reverse:
                return True
            else:
                return False

File: C#/9.cs
public class Solution {
    public bool IsPalindrome(int x) {
        if(x < 0) {
            return false; // negative numbers are not palindromes
        }
        int reversed = 0;
        int original = x;
        while(x != 0) {
            int digit = x % 10;
            reversed = reversed * 10 + digit;
            x /= 10;
        }
        return reversed == original; // check if the reversed number is equal to the original number
    }
}